Add Landing Zone References to the website (#275)
gruntwork-io/gruntwork-io.github.io: _posts/2019-08-12-how-to-configure-production-grade-aws-account-structure.adoc

== Intro

This guide will walk you through the process of configuring a production-grade AWS account structure using the Gruntwork

> Landing Zone solution, including how to manage multiple environments, users, permissions, audit logging, and more.

Reviewer: We only talk about Gruntwork's Landing Zone solution in the deployment walkthrough, so for now, introduce the concept more generically:

```suggestion
This guide will walk you through the process of configuring a production-grade AWS account structure, including how to manage multiple environments, users, permissions, and audit logging. We'll also discuss how to implement a _Landing Zone_ solution that lets you quickly spin up new AWS accounts that all implement a _security baseline_ that enforces your company's policies. 
```

  link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library#production_grade_infra_checklist[The production-grade infrastructure checklist].

<<deployment_walkthrough>>::
  A step-by-step guide to configuring a production-grade AWS account structure using the Gruntwork AWS Landing Zone

>   solution including how to manage it all with customizable security baselines defined in Terraform.

Reviewer: ```suggestion
  solution, including how to manage it all with customizable security baselines defined in Terraform.
```

<<next_steps>>::
  What to do once you've got your AWS account structure configured.

Feel free to read the guide from start to finish or skip around to whatever part interests you.

=== Gruntwork AWS Landing Zone vs the competition

When we evaluated the market, we found a number of pre-built "Landing Zone" solutions that were either expensive,
low-quality, UI-driven, and/or didn't allow customization. The Gruntwork AWS Landing Zone solution allows you to
set up production-grade AWS accounts out of the box in minutes, but it's also 100% defined as code, so you can
fully customize it to your needs. Our solution follows a "Terraform-native" approach that doesn't require you to
install anything extra and uses Terraform to deploy and configure all of the new AWS accounts. It makes it easy
to customize your landing zone configuration using our example security baseline modules or by leveraging
Gruntwork's upstream updates. Under the hood, our new security baseline modules use a variety of other modules

> such as our GuardDuty module, AWS Config, IAM roles, IAM groups, and so on.

Author: Thanks Josh, I've taken it nearly verbatim and only tweaked it slightly. 👍

multi-region feature enabled, as the API call data is useful useful for troubleshooting, investigating security
incidents, and maintaining audit logs for compliance.

=== AWS Config

_https://aws.amazon.com/config/[AWS Config]_ is a service that enables you to assess, audit, and evaluate the configurations of
your https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html[AWS resources]. You can use AWS
Config to ensure that AWS resources are configured in a manner that is in compliance with your company policies or regulatory
requirements. This enables you to simplify compliance auditing, security analysis, change management, and operational
troubleshooting.

> AWS Config Rules::

Reviewer: NIT: since there's only one item here, using a definition list doesn't make sense. Perhaps just transition into this from the previous paragraph via, "The way you specify your company's policies and requirements is by defining _AWS Config Rules_, which are expressions..."

Author: Ok fixed.

give a small number of trusted admins access to the KMS master key and the S3 bucket. You may also want to send the
logs to CloudWatch Logs a second way to store/view audit logs.

=== AWS Config

> You'll want to enable AWS Config in all available regions in each of your AWS accounts so that you can track AWS

Reviewer: The fact that you enable Config in every region is probably non-obvious and should be called out more. E.g., "You'll want to enable AWS Config in every one of your AWS accounts, plus any AWS Config Rules that are relevant to your company, so that you can track AWS resource configuration changes over time and enforce company policy. Note that you should enable AWS Config _in every AWS region_, and not just the region(s) you're using for the rest of your infrastructure That way, if an employee (perhaps accidentally), or even worse, an attacker, runs something in a region you don't typically use, it'll still be recorded in AWS Config, and your rules will still be enforced."

Same thing applies to the GuardDuty section below!

Author: Ok I've updated both.

give a small number of trusted admins access to the KMS master key and the S3 bucket. You may also want to send the
logs to CloudWatch Logs a second way to store/view audit logs.

=== AWS Config

You'll want to enable AWS Config in every one of your AWS accounts, plus any AWS Config Rules that are relevant to your
company, so that you can track AWS resource configuration changes over time and enforce company policy. Note that you
should enable AWS Config in every AWS region, and not just the region(s) you're using for the rest of your infrastructure

> that way, if an employee (perhaps accidentally), or even worse, an attacker, runs something in a region you don't typically

Reviewer: ```suggestion
should enable AWS Config in every AWS region, and not just the region(s) you're using for the rest of your infrastructure.
That way, if an employee (perhaps accidentally), or even worse, an attacker, runs something in a region you don't typically
```

use, it'll still be recorded in AWS Config, and your rules will still be enforced. We typically recommend creating a separate
S3 bucket for each region and storing all of the configuration items there for a set number of days, after which they should
be moved to Glacier.

=== GuardDuty

You'll want to enable GuardDuty in every one of your AWS accounts, so that you can protect both your accounts and workloads
against malicious activity and unauthorized behavior. Note that you should enable GuardDuty in every AWS region, and not just

> the region(s) you're using for the rest of your infrastructure that way, if an employee (perhaps accidentally), or even worse,

Reviewer: ```suggestion
the region(s) you're using for the rest of your infrastructure. That way, if an employee (perhaps accidentally), or even worse,
```

[[deployment_walkthrough]]
== Deployment walkthrough

Let's now walk through the step-by-step process of how to create a production-grade AWS account structure using the
Gruntwork AWS Landing Zone solution. Doing this from scratch is time-consuming and complicated, and all the pre-built
"Landing Zone" solutions we tried were expensive, low-quality, UI-driven, and/or didn't allow customization. The
Gruntwork AWS Landing Zone solution allows us to do this in minutes, but it's also 100% defined as code, so you can
fully customize it to your needs. The best part is that it automatically configures your accounts with a variety of
authentication, access controls and security features by using AWS Organizations, IAM Roles, IAM Users, IAM Groups,
IAM Password Policies, Amazon GuardyDuty, AWS CloudTrail, AWS Config, and a whole lot more so we can avoid doing

> this manually.

Reviewer: You already introduced it in the Core Concepts section, so this is a 100% repeat. I'd reduce this back down to a single sentence:

> Let's now walk through the step-by-step process of how to create a production-grade AWS account structure using the Gruntwork AWS Landing Zone solution.

Author: Ok, roger.

group (solely for a few trusted admins), plus a number of groups that give specific permissions in all your other child accounts (e.g., full-access in dev,
read-only in prod, etc). Create an IAM user for yourself in the `full-access` group, plus IAM users for the rest of your team in the appropriate groups.

Like the root account, the code will also generate a password for each user and encrypt it with that user’s PGP key from Keybase (see below for how to handle
the passwords). You should follow this pattern to create an IAM user for yourself, as well as the small number of other trusted admins at your company who should
have access to the root account.

Configure your Terraform backend:

.infrastructure-live/security/_global/account-baseline/terragrunt.hcl
[source,hcl]
----
include {
  path = find_in_parent_folders()
}
----

https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html#id_root-user_manage_add-key[Create a set of access keys for the root user in the security account]
and https://blog.gruntwork.io/a-comprehensive-guide-to-authenticating-to-aws-on-the-command-line-63656a686799[use those access keys to authenticate on the CLI].

> Finally, deploy the `iam` module by running `terragrunt apply`:

Reviewer: ```suggestion
Finally, deploy the `account-baseliine` module by running `terragrunt apply`:
```

Author: good find, but changing to `account-baseline`.

terragrunt apply
----

> === Enable AWS Config in the root account

Reviewer: Rob, just took a quick glance and noticed you are updating the guide to deploy each module, one at a time, but we should instead replace the deployment section entirely with a recommendation to use the new account-baseline modules! That would reduce the entire guide down to solely deploying the three account-baseline modules, turning on/off whatever settings the customer wants.

Author: @brikis98 I would have done but according to your PRD it wasn't a "must-have".

Reviewer: It wasn't a must-have because the account-baseline modules weren't must-haves... But since we did them, it only makes sense to update the guide to use them!

Author: @brikis98 okay I'm really eager to try and finish this, this week, which means today, so I would appreciate a further review of the rest of the stuff I've done so far before I start making changes.

Reviewer 2: +1 on Jim's suggestion to use the account-baseline modules. In fact, I show those in most of my sales demos these days and customers always respond very favorably to them. It will also present the setup process as less complex, which is a key driver for customers.

Author: I've updated the guide to use the new `account-baseline` modules.
